### ISSUE TITLE
Default action not executed in Selectrum

### ISSUE THREAD

**Issue by minad** [oantolin/embark#64]
@oantolin I decided to create another issue for the RET behavior, extracted from #62.

I am pressing the keys `M-x consult-line M-SPC RET`. In my config `M-SPC` is bound to `embark-act-noexit`.

The expected behavior would be that `RET` executes the default action and performs the jump to the line. This works with icomplete. But Selectrum behaves differently. Instead the minibuffer shows `Go to line: [some full line candidate]`. Then I have to press `<down>` and `RET` to select the candidate and to perform the jump.

Since this is Selectrum-related, ping @clemera .

**Comment by oantolin** [comment]
Thanks for filing this as a separate issue. I'm not sure how much help I'll be able to give on this one, at least not until I start learning more about selectrum.

**Comment by minad** [comment]
@oantolin I don't think this is a deep issue, just some keybinding issue or how the injection mechanism triggers the execution via unread-command-events. I think this issue should have low prio.

**Comment by oantolin** [comment]
> how the injection mechanism triggers the execution via unread-command-events

This sounds pretty likely. I can't really remember why it is done that way.

**Comment by minad** [comment]
Well, I think it is the standard mechanism to inject key presses - seems like the natural thing to use, I guess?

**Comment by oantolin** [comment]
But why not call `exit-minibuffer`, say? `RET` could do anything! Embark relies on the user binding `RET` to something not too crazy.

**Comment by minad** [comment]
Probably `exit-minibuffer` would work too and could be less problematic.

**Comment by oantolin** [comment]
Try the update, now I'm scheduling exit-minibuffer to run, instead of putting `RET` in the input queue.

**Comment by minad** [comment]
@oantolin It seems to work like before. But I think the issues lies somewhere else, it seems the consult-preview-mode is what triggers the slight difference in behavior. I will investigate this.

**Comment by minad** [comment]
Do you think using exit-minibuffer is more robust than 13? Probably makes sense to keep it like this?

**Comment by oantolin** [comment]
Yes, I'll keep the exit minibuffer version, it makes much more sense.

**Comment by minad** [comment]
I checked this bug again after the refactoring, and now something different happens.

`M-x consult-line M-SPC RET` -> Error: Consult called from minibuffer. M-SPC is my embark-act-no-exit binding. It works well for embark-act-exit.

**Comment by oantolin** [comment]
Now that I have been a Selectrum user for almost an hour, I'll try to debug this.

**Comment by oantolin** [comment]
I can't reproduce this in my configuration. I'll try in `emacs -q`.

**Comment by oantolin** [comment]
I can't reproduce this in `emacs -q` either. Maybe try again after updating?

**Comment by minad** [comment]
@oantolin Same here, I will try to give you an update on this issue soon, if it still exists.

**Comment by minad** [comment]
Also closing this one, I think embark changed too much since when I observed this. There is still a new/similar issue, I will create a new issue for that to restart the discussion.
